plugin/sdk_v2/shared/core/bus_context.py

```
@@ -0,0 +1,781 @@
+            return True
+
+        return SdkBusList(
+            [item for item in self.items if _matches(item)],
+            namespace=self._namespace,
+            record_factory=self._record_factory,
+            host_ctx=self._host_ctx,
+        )
+
+    def filter(self, flt: object | None = None, *, strict: bool = True, **kwargs: Any) -> "SdkBusList[TRecord]":
+        if flt is not None and not callable(flt):
+            raise TypeError("filter predicate must be callable")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)